### DISCUSSION TITLE
[@types/yargs] type field regression on option

### DISCUSSION THREAD

**Discussion by idoros** [DefinitelyTyped/DefinitelyTyped#63433] [Issues with a @types package]
There seems to be a regression in `yargs@17.0.15` (seems to originate [here](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/63205/files#diff-e1c9bcedd5cf08cf3e697aef9539535d9a16d0e92f33c9a5038b3389e6490c0cR851)), the new version creates a union of the explicit type of the default field with the type field type (`(string | number)[]` for "array")

For example:
```ts
const argv = yargs.option('require', {
    type: 'array',
    default: [] as string[],
});
```
This used to resolve to:

```
yargs.Argv<{
    require: string[];
}>
```

And now resolves to:
```
yargs.Argv<{
    require: string[] | (string | number)[];
}>
```

This can be fixed with by adding `string: true` to the option, but it still requires the `as string[]`, to avoid getting `never[] | string[]`:
```ts
const argv = yargs.option('require', {
    type: 'array',
    string: true,
    default: [] as string[],
});
```

[a comment by an automated account is not shown]

**Comment by shadowspawn** [comment]
The number type is accurate as setting a default value of a string array does not change the command-line parsing turning numbers into numbers by default.

For example, logging the parse result from your code:

```console
$ node index --require 1 --require 2 --require str
{ _: [], require: [ 1, 2, 'str' ], '$0': 'index' }
```

**Comment by idoros** [comment]
Fair enough, so `string: true` is used to force the casting in runtime, but the type of default could maybe be dropped (`default: [] as string[],`)?
